veles/secrets/gcshmackey/validator.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcshmackey
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials"
```

Comment:
I suppose there's a non-trivial amount of code in the SDK that we need to be able to validate to be able to run the validation without importing this library?

Importing new libs internally is a bit of a hassle so if we're only using a small subset of the functionality from here (e.g. sending a simple HTTP request to an endpoint) we could consider adding that code ourselves.